internal/storage/v2/grpc/trace_storage.proto

```diff
@@ -83,4 +123,27 @@ service TraceReader {
   // GetOperations returns all operation names for a given service
   // known to the backend from traces within its retention period.
   rpc GetOperations(GetOperationsRequest) returns (GetOperationsResponse) {}
+
+  // FindTraces returns a stream that retrieves traces matching query parameters.
+  //
+  // The chunking rules are the same as for GetTraces.
+  //
+  // If no matching traces are found, an empty stream is returned.
+  rpc FindTraces(FindTracesRequest) returns (stream opentelemetry.proto.trace.v1.TracesData) {}
+
+  // FindTraceIDs returns a stream that retrieves IDs of traces matching query parameters.
+  //
+  // If no matching traces are found, an empty stream is returned.
+  //
+  // This call behaves identically to FindTraces, except that it returns only the list
+  // of matching trace IDs. This is useful in some contexts, such as batch jobs, where a
+  // large list of trace IDs may be queried first and then the full traces are loaded
+  // in batches.
+  rpc FindTraceIDs(FindTracesRequest) returns (FindTraceIDsResponse) {}
+}
+
+// The TraceWriter is used for pushing a batch of traces to the storage backend.
+// This service is comptaible with OTEL's TraceService.
+service TraceWriter {
```

Comment:
Why wouldn't we just use OTEL export API? it would make any OTLP compatible receiver also compatible with Jaeger as a write destination. If we redeclare it we introduce a new service name which isn't compatible.